tests/rkt_image_list_test.go

```diff
@@ -0,0 +1,178 @@
+// command is usable by the commands that accept image hashes.
+func TestShortHash(t *testing.T) {
+	var (
+		imageIds []ImageId
+		iter     int
+	)
+
+	for {
+		image := patchTestACI(fmt.Sprintf("rkt-shorthash-%d.aci", iter), fmt.Sprintf("--name=shorthash--%d", iter))
+		defer os.Remove(image)
+
+		imageHash := getHashOrPanic(image)
+		imageId := ImageId{image, imageHash}
+
+		imageIdPair, isMatch := imageId.containsConflictingHash(imageIds)
+		if isMatch {
+			imageIds = imageIdPair
+			break
+		} else {
+			imageIds = append(imageIds, imageId)
+			iter++
+		}
+	}
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+
+	// Pull images into cas
+	for _, imageId := range imageIds {
+		cmd := fmt.Sprintf("%s --insecure-skip-verify fetch %s", ctx.cmd(), imageId.path)
+		t.Logf("Fetching %s: %v", imageId.path, cmd)
+		child, err := gexpect.Spawn(cmd)
+		if err != nil {
+			t.Fatalf("Cannot exec: %v", err)
+		}
+		if err := child.Wait(); err != nil {
+			t.Fatalf("rkt didn't terminate correctly: %v", err)
+		}
+	}
+
+	// Get hash from 'rkt image list'
+	imageListCmd := fmt.Sprintf("%s image list --fields=key --no-legend", ctx.cmd())
+	child, err := gexpect.Spawn(imageListCmd)
+	if err != nil {
+		t.Fatal("Cannot exec rkt image list")
+	}
+
+	hash := fmt.Sprintf("sha512-%s", imageIds[0].hash[:12])
+	if err = expectWithOutput(child, hash); err != nil {
+		t.Fatalf("Couldn't find %s in: %v", hash, err)
+	}
+
+	tmpDir := createTempDirOrPanic("rkt_image_list_test")
+	defer os.RemoveAll(tmpDir)
+
+	// Define tests
+	tests := []struct {
+		cmd    string
+		expect string
+	}{
+		{
+			"image cat-manifest sha512-12341234",
+			"no keys found",
+		},
+		{
+			fmt.Sprintf("image cat-manifest %s", fmt.Sprintf("sha512-%s", imageIds[1].hash[:1])),
+			"key too short",
+		},
+		{
+			fmt.Sprintf("image cat-manifest %s", fmt.Sprintf("sha512-%s", imageIds[1].hash[:2])),
+			"ambiguous key",
+		},
+		{
+			fmt.Sprintf("image cat-manifest %s", hash),
+			"ImageManifest",
+		},
+		{
+			fmt.Sprintf("image export --overwrite %s %s/export.aci", hash, tmpDir),
+			"",
+		},
+		{
+			fmt.Sprintf("image render --overwrite %s %s", hash, tmpDir),
+			"",
+		},
+		{
+			fmt.Sprintf("image extract --overwrite %s %s", hash, tmpDir),
+			"",
+		},
+		{
+			fmt.Sprintf("prepare %s", hash),
+			"Writing pod manifest",
+		},
+		{
+			fmt.Sprintf("image rm %s", hash),
+			"cannot be removed",
+		},
+		{
+			fmt.Sprintf("image rm %s", fmt.Sprintf("sha512-%s", imageIds[1].hash[:12])),
+			"successfully removed aci",
+		},
+	}
+
+	// Run tests
+	for _, tt := range tests {
+		runCmd := fmt.Sprintf("%s %s", ctx.cmd(), tt.cmd)
+		t.Logf("executing: %v", runCmd)
+		child, err = gexpect.Spawn(runCmd)
+
+		if tt.expect != "" {
+			if err = expectWithOutput(child, tt.expect); err != nil {
+				t.Fatalf("Expected but didn't find %q in %v", tt.expect, err)
+			}
+		}
+
+		if err = child.Wait(); err != nil {
+			if err.Error() != "exit status 1" {
```

Comment:
It should not accept "exit status 1" for all commands but instead only those who are supposed to fail.